src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchService.java

```diff
@@ -103,33 +99,44 @@ OffenderMatchDetail getOffenderMatchDetail(String crn) {
-                            .onErrorResume(OffenderNotFoundException.class, e -> Mono.just(Collections.emptyList())),
+                                .onErrorResume(OffenderNotFoundException.class, e -> Mono.just(Collections.emptyList())),
                         offenderRestClient.getProbationStatusByCrn(crn))
-            .map(tuple -> addMostRecentEventToOffenderMatch(tuple.getT1(), tuple.getT2(), tuple.getT3()))
-            .block();
+                .map(tuple -> addMostRecentEventToOffenderMatch(tuple.getT1(), tuple.getT2(), tuple.getT3()))
+                .block();
     }
 
     private OffenderMatchDetail addMostRecentEventToOffenderMatch(OffenderMatchDetail offenderMatchDetail,
-                                                          List<Conviction> convictions,
-                                                          ProbationStatusDetail probationStatus) {
+                                                                  List<Conviction> convictions,
+                                                                  ProbationStatusDetail probationStatus) {
 
         if (offenderMatchDetail == null) {
             return null;
         }
 
         Sentence sentence = Optional.ofNullable(convictions).orElse(Collections.emptyList()).stream()
-            .filter(Conviction::getActive)
-            .findFirst()
-            .map(Conviction::getSentence)
-            .orElse(getSentenceForMostRecentConviction(convictions));
+                .filter(Conviction::getActive)
+                .findFirst()
+                .map(Conviction::getSentence)
+                .orElse(getSentenceForMostRecentConviction(convictions));
 
         return OffenderMapper.offenderMatchDetailFrom(offenderMatchDetail, sentence, probationStatus);
     }
 
     Sentence getSentenceForMostRecentConviction(List<Conviction> convictions) {
         return Optional.ofNullable(convictions).orElse(Collections.emptyList()).stream()
-            .min(Comparator.comparing(Conviction::getConvictionDate, Comparator.nullsLast(Comparator.reverseOrder())))
-            .map(Conviction::getSentence)
-            .orElse(null);
+                .min(Comparator.comparing(Conviction::getConvictionDate, Comparator.nullsLast(Comparator.reverseOrder())))
+                .map(Conviction::getSentence)
+                .orElse(null);
+    }
+
+    public Mono<GroupedOffenderMatchesEntity> getGroupedOffenderMatchesByDefendantIdAndGroupId(String defendantId, Long groupId) {
+        log.debug("Entered getGroupedMatchesByDefendantId with defendantId {} , groupId {}", defendantId, groupId);
+        return Mono.justOrEmpty(offenderMatchRepository.findById(groupId))
+                .map(groupedOffenderMatchesEntity -> {
+                    if (!defendantId.equals(groupedOffenderMatchesEntity.getDefendantId())) {
+                        throw new EntityNotFoundException(String.format("Grouped Matches %s not found for defendant %s", groupId, defendantId));
```

Comment:
You could do this with a filter so that it consistently returns a Mono.empty() if it's not found. The net effect will be the same but it might be more idiomatic 🤷  something like:
```
.filter(groupedOffenderMatchesEntity -> defendantId.equals(groupedOffenderMatchesEntity.getDefendantId())
```
It reads well either way, arguably yours reads better in fact 🤔  Up to you!

Reply:
Thanks for the tip, I am new to  RX and will try to use the suggestion next time.